scan.go

```diff
@@ -238,6 +239,16 @@ func scan(pkgPatterns []string) ([]*unit.SourceUnit, error) {
 		// compare results in tests more easily.
 		pkg.Root = ""
 
+		if strings.HasPrefix(pkg.ImportPath, "_") {
+			wd, err := os.Getwd()
+			if err != nil {
+				return nil, err
+			}
+			pkg.ImportPath = repo + pkg.ImportPath[1+len(wd):]
```

Comment:
It seems fragile to modify `pkg` here. For example, what if some other field in `pkg` needs to change (potentially one that gets added down the road in the future). An alternative approach would be:

1. Detect if we're outside the GOPATH
2. Modify GOPATH to include `srclibGopath`
3. run `scan(pkgPatterns)` as normal with the modified GOPATH

I.e., the `scan` function is not responsible for detecting whether we're outside the GOPATH. That's done at the top leve of `ScanCmd`

This way, if `pkg` is created successfully, we have much more confidence in the correctness of its fields than if we modified them ourselves.